ACE/ace/config-g++-common.h

```diff
@@ -38,7 +38,22 @@
 # if __cplusplus > 201103L
 #  define ACE_HAS_CPP14
 # endif
-#endif
+#endif /* __GNUC__ >= 4.7 */
+
+// *NOTE*: this feature may go back further, see e.g.:
+//         https://gcc.gnu.org/projects/cxx0x.html
+#if defined (ACE_HAS_CPP11)
+# if (__GNUC__ >= 5 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 3))
+#  define ACE_HAS_CPP11_EXTERN_TEMPLATES
+# endif /* __GNUC__ >= 4.3 */
+
+// *NOTE*: suppress a warning, g++ 5.2.1 does not support attributes on template
+// instantiation declarations
+// *TODO*: this probably goes back further than 5.2
+# if (__GNUC__ >= 6 || (__GNUC__ == 5 && __GNUC_MINOR__ >= 2))
+#  define ACE_LACKS_CPP11_EXTERN_TEMPLATE_ATTRIBUTES
```

Comment:
this change needs to be tested against newer versions of gcc (I don't have gcc >= 6), and disabled when it integrates `extern template` attributes in the future. Also, this may need adjustment, as the patch applies to earlier versions of gcc as well (probably as far back as 4.3, when `ACE_HAS_CPP11_EXTERN_TEMPLATES` was introduced).